File: solutions/julia/pascals-triangle/1/pascals-triangle.jl
function triangle(n)
    if n < 0
        throw(DomainError(n))
    end
    ans = []
    for i in 1:n
        line = [1]
        for j in 2:(i-1)
            push!(line, ans[i-1][j-1] + ans[i-1][j])
        end
        if i > 1
            push!(line, 1)
        end
        push!(ans, line)
    end
    ans
end

File: solutions/kotlin/pascals-triangle/1/src/main/kotlin/PascalsTriangle.kt
object PascalsTriangle {

    fun computeTriangle(rows: Int): List<List<Int>> {
        require(rows >= 0)
        val ans: MutableList<List<Int>> = mutableListOf()
        for (i in 0..<rows) {
            val line: MutableList<Int> = mutableListOf(1)
            for (j in 1..<i) {
                line.add(ans[i-1][j-1] + ans[i-1][j])
            }
            if (i > 0) {
                line.add(1)
            }
            ans.add(line)
        }
        return ans
    }
}
